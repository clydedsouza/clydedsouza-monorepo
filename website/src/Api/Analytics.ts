import mixpanel from 'mixpanel-browser'
import { PageTypes } from '../Components/Navigation/PageTypes'
import {
  ILinkClickedProps,
  MIXPANEL_DEV_PROJECT_ID,
  MIXPANEL_PROD_PROJECT_ID,
} from './IAnalytics'

export const initAnalyticsWithSuperProperties = () => {
  const mixpanelProjectId =
    process.env.NODE_ENV === 'production'
      ? MIXPANEL_PROD_PROJECT_ID
      : MIXPANEL_DEV_PROJECT_ID

  try {
    mixpanel.init(mixpanelProjectId)
    mixpanel.register({
      environment: process.env.NODE_ENV,
    })
  } catch {
    undefined
  }
}

export const sendPageViewEvent = (pageType: PageTypes) => {
  try {
    mixpanel.track('Page view', { page: pageType })
  } catch {
    undefined
  }
}

export const sendLinkClickedEvent = (props: ILinkClickedProps) => {
  try {
    mixpanel.track('Link clicked', {
      destination_url: props.link,
      link_type: props.type,
      page_location: props.location,
    })
  } catch {
    undefined
  }
}
