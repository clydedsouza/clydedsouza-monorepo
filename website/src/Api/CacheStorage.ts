import { PageTypes } from '../Components/Navigation/PageTypes'
import { IProjectData } from './IProjectData'

export const projectDataCacheMap: Map<PageTypes, IProjectData> = new Map()
