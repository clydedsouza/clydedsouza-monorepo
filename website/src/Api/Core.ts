import axios from 'axios'
import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'

const BASE_URL = 'https://api.clydedsouza.net/'

const getProjectApiUrl = (pageType: PageTypes) => {
  switch (pageType) {
    case PageTypes.Books:
      return `${BASE_URL}books.json`
    case PageTypes.Speaking:
      return `${BASE_URL}speaking.json`
    case PageTypes.Teaching:
      return `${BASE_URL}teaching.json`
    case PageTypes.Projects:
    default:
      return `${BASE_URL}pinned.json`
  }
}

export const getProjectData = (pageType: PageTypes): Promise<IProjectData> => {
  const request = getProjectApiUrl(pageType)
  return axios
    .get(request)
    .then((response) => {
      return response.data
    })
    .catch(() => {
      throw new Error('API error')
    })
}
