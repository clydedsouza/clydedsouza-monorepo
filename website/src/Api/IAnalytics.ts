export const MIXPANEL_DEV_PROJECT_ID = '6c905af7861c30571d76bbc6d67696ef'
export const MIXPANEL_PROD_PROJECT_ID = '50f70ae3a52dd646aea4025dc1f30a06'

export enum AnalyticsLinkType {
  Cta = 'Cta',
  SocialIcons = 'Social icons',
  CardGitHub = 'GitHub',
  CardWebsite = 'Website',
  GeneralWebsite = 'External website',
}

export interface ILinkClickedProps {
  link?: string
  type?: AnalyticsLinkType
  location?: string
}
