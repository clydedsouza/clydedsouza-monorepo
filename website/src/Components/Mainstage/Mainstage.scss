@import '../../Sass/variables';

main {
  width: 64%;
  padding: 16px 35px;
  display: inline-block;
  padding-left: 34%;

  @include for-tablets-down {
    & {
      padding: 16px;
      height: auto;
      display: inline-block;
    }
  }

  @include for-tablets-down {
    padding-left: 21px;
  }
}
