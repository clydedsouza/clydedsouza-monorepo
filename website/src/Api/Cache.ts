import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'
import { getProjectData } from './Core'

const projectDataCacheMap: Map<PageTypes, Promise<IProjectData>> = new Map()

export const getCachedProjectData = (
  pageType: PageTypes
): Promise<IProjectData> => {
  if (!projectDataCacheMap.has(pageType)) {
    const getProjectDataPromise = getProjectData(pageType).catch((error) => {
      projectDataCacheMap.delete(pageType)
      return Promise.reject(error)
    })
    projectDataCacheMap.set(pageType, getProjectDataPromise)
  }

  return projectDataCacheMap.get(pageType)!
}
