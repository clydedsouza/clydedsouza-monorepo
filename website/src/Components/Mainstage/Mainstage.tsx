import Footer from '../Footer/Footer'
import Navigation from '../Navigation/Navigation'
import './Mainstage.scss'

function Mainstage() {
  return (
    <main>
      <Navigation />
      <Footer />
    </main>
  )
}

export default Mainstage
