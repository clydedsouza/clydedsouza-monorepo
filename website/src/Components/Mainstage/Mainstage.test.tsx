import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import * as apiCacheModule from '../../Api/Cache'
import Mainstage from './Mainstage'

jest.mock('../../Api/Cache')

const apiDataMockResponse = [
  {
    id: 'ai-tell-me-a-story',
    contents: '',
    title: 'Purchase my book AI, Tell Me a Story from these platforms.',
    hasWebsite: true,
    website: 'https://aitellmeastory.clydedsouza.net',
    excerpt: '',
  },
]

describe('Mainstage', () => {
  beforeEach(() => {
    jest.spyOn(apiCacheModule, 'getCachedProjectData').mockImplementation(() =>
      Promise.resolve({
        app: {},
        data: [apiDataMockResponse[0]],
      })
    )
  })

  it('should render mainstage', async () => {
    render(<Mainstage />)
    await waitForElementToBeRemoved(() =>
      screen.queryAllByText('Contents are loading...')
    )
    expect(document.body).toMatchSnapshot()
  })
})
