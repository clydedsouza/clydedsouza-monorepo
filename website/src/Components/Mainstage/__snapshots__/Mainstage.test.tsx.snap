// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainstage should render mainstage 1`] = `
<body>
  <div>
    <main>
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          About
        </a>
        <a
          class=""
          href="/portfolio"
        >
          Portfolio
        </a>
        <a
          class=""
          href="/platforms"
        >
          Platforms
        </a>
      </nav>
      <h2>
        About me
      </h2>
      <p>
        Hi! I'm Clyde D'Souza. I'm a creative and self-driven person who likes to code and do other things. I'm currently working as a senior software engineer at
         
        <a
          href="https://www.xero.com/nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Xero
        </a>
        . Previously, I've worked with
         
        <a
          href="https://www.datacom.co.nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Datacom
        </a>
         
        and
         
        <a
          href="https://www.heritagehotels.co.nz/?ref=clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Heritage Hotels
        </a>
        .
      </p>
      <p>
        I was the event director of a non-profit educational event called Light & Spark NPO. I conceptualized and executed this event in Mumbai along with my
         
        <a
          href="https://lightandsparknpo.github.io/team/"
          rel="noreferrer"
          target="_blank"
        >
          amazing team
        </a>
        . Read more about it
         
        <a
          href="https://lightandsparknpo.github.io/2019-11-20-ruia-mumbai-2018/"
          rel="noreferrer"
          target="_blank"
        >
          here
        </a>
        .
      </p>
      <p>
        I've published a book titled Mama, Tell Me a Story which is available on
         
        <a
          href="http://mamatellmeastory.clydedsouza.net/"
          rel="noreferrer"
          target="_blank"
        >
          these platforms
        </a>
        . It's a collection of twelve short bedtime stories that parents will love reading to their kids over and over again. Go buy it now!
      </p>
      <p>
        In my spare time, I like to explore other interests like
         
        <a
          href="https://medium.com/@clydedz"
          rel="noreferrer"
          target="_blank"
        >
          writing
        </a>
        ,
         
        <a
          href="https://www.behance.net/clydedz"
          rel="noreferrer"
          target="_blank"
        >
          designing
        </a>
        ,
         
        <a
          href="https://www.skillshare.com/r/user/clydedsouza"
          rel="noreferrer"
          target="_blank"
        >
          teaching
        </a>
        , and I've also created many
         
        <a
          href="/#/projects"
        >
          other projects
        </a>
        .
      </p>
      <p>
        I'm an alumnus of
         
        <a
          href="https://www.aut.ac.nz/?ref=clydedsouza"
          rel="noreferrer"
          target="_blank"
        >
          Auckland University of Technology (AUT)
        </a>
         
        with a Postgraduate Diploma in Computer and Information Sciences and holds a Bachelor of Science degree specializing in Information Technology from University of Mumbai, India.
      </p>
      <br />
      <h2>
        Highlights
      </h2>
      <div
        class="cards"
      >
        <div
          class="card"
        >
          <div
            aria-label=""
          >
            <span />
          </div>
          <div>
            <h3>
              Purchase my book AI, Tell Me a Story from these platforms.
            </h3>
            <p />
            <div
              class="card-links"
            >
              <a
                href="https://aitellmeastory.clydedsouza.net"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="globe-svg"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 256c0 22.2-1.2 43.6-3.3 64H163.3c-2.2-20.4-3.3-41.8-3.3-64s1.2-43.6 3.3-64H348.7c2.2 20.4 3.3 41.8 3.3 64zm28.8-64H503.9c5.3 20.5 8.1 41.9 8.1 64s-2.8 43.5-8.1 64H380.8c2.1-20.6 3.2-42 3.2-64s-1.1-43.4-3.2-64zm112.6-32H376.7c-10-63.9-29.8-117.4-55.3-151.6c78.3 20.7 142 77.5 171.9 151.6zm-149.1 0H167.7c6.1-36.4 15.5-68.6 27-94.7c10.5-23.6 22.2-40.7 33.5-51.5C239.4 3.2 248.7 0 256 0s16.6 3.2 27.8 13.8c11.3 10.8 23 27.9 33.5 51.5c11.6 26 20.9 58.2 27 94.7zm-209 0H18.6C48.6 85.9 112.2 29.1 190.6 8.4C165.1 42.6 145.3 96.1 135.3 160zM8.1 192H131.2c-2.1 20.6-3.2 42-3.2 64s1.1 43.4 3.2 64H8.1C2.8 299.5 0 278.1 0 256s2.8-43.5 8.1-64zM194.7 446.6c-11.6-26-20.9-58.2-27-94.6H344.3c-6.1 36.4-15.5 68.6-27 94.6c-10.5 23.6-22.2 40.7-33.5 51.5C272.6 508.8 263.3 512 256 512s-16.6-3.2-27.8-13.8c-11.3-10.8-23-27.9-33.5-51.5zM135.3 352c10 63.9 29.8 117.4 55.3 151.6C112.2 482.9 48.6 426.1 18.6 352H135.3zm358.1 0c-30 74.1-93.6 130.9-171.9 151.6c25.5-34.2 45.2-87.7 55.3-151.6H493.4z"
                  />
                </svg>
              </a>
              <a
                href="https://github.com"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  class="github-svg"
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <footer>
        <div
          class="cta"
          data-location="footer"
        >
          <a
            href="https://aitellmeastory.clydedsouza.net"
            rel="noreferrer"
            target="_blank"
          >
            Purchase my book AI, Tell Me a Story from these platforms.
          </a>
        </div>
        <p>
          Website crafted by Clyde DSouza
        </p>
      </footer>
    </main>
  </div>
</body>
`;
